src/Nancy/Extensions/StreamExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace Nancy.Extensions
+{
+    using System.IO;
+    using System.Text;
+
+    /// <summary>
+    /// Extensions for Stream.
+    /// </summary>
+    public static class StreamExtensions
+    {
+        /// <summary>
+        /// Gets the request body as a string.
+        /// </summary>
+        /// <param name="stream">The request body stream.</param>
+        /// <param name="encoding">The encoding to use, <see cref="Encoding.UTF8"/> by default.</param>
+        /// <returns>The request body as a <see cref="string"/>.</returns>
+        public static string AsString(this Stream stream, Encoding encoding = null)
+        {
+            var reader = new StreamReader(stream, encoding ?? Encoding.UTF8);

```

Comment:
We should pass `true` for `leaveOpen` so the stream is left open and keep disposing the reaser.